### ISSUE TITLE
Difference in __total__ TypedDict using NotRequired

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [python/cpython#105074]
I'm unsure if this is really a bug, but it's behavior that surprised me upgrading a typed dict to use Required/NotRequired 655 style syntax so would help to clarify.

# Bug report

```python
class _MovieBase(TypedDict):  # implicitly total=True
    title: str

class Movie1(_MovieBase, total=False):
    year: int
```

vs 

```python
class Movie2(TypedDict):
    title: str
    year: NotRequired[int]
```

is documented in the pep as more readable replacement. There is one way though that new syntax behaves differently then old that caused some tests I had to fail. (edited typo here) For`Movie1.__total__` is false. For `Movie2.__total__` is true. This is surprising when expecting `__total__` to mean all keys required. The current documentation does note [though](https://docs.python.org/3/library/typing.html#typing.TypedDict.__total__) `Point2D.__total__ gives the value of the total argument. Example:`. That documentation does exactly match implementation [here](https://github.com/python/cpython/blob/d593074494b39a3d51e67a4e57189c530867a7ff/Lib/typing.py#L2808) which looks to be true for years.

# Your environment

Tested on python 3.9 (using typing-extensions TypedDict/NotRequired), but reading main total argument is used for `__total__` without considering optional keys so should apply to main as well. The relevant line hasn't changed in ~4 years either so I expect it to reproduce for all versions since NotRequired (3.11+).

**Comment by JelleZijlstra** [comment]
> For`Movie1.__total__` is true. For `Movie2.__total__` is false. 

I think you mean the opposite.

> This is surprising when expecting `__total__` to mean all keys required.

That expectation is wrong, as the `Point3D` example in the docs makes clear.

I'm not sure what change you are asking for, but it's probably worth making the docs for `__total__` more explicit that the value isn't all that meaningful in the presence of inheritance and `Required`/`NotRequired`: you should use `__required_keys__` and `__optional_keys__` instead.

**Comment by hmc-cs-mdrissi** [comment]
Yes I got Movie1/Movie2 backwards writing that up.

I think a recommendation of prefer `__required_keys__`/`__optional_keys__` makes sense and is how I fixed my bug. Original code was written in 3.8 before them, but for any typed dict where Required may exist `total` now seems of low value.

**Comment by sobolevn** [comment]
I agree with @JelleZijlstra, `__total__` has a documented behaviour: 

> __total__ gives the value of the total argument.

I also agree that the docs can be improved:
1. I think we can add an explicit example with `total=True` and `NotRequired` to show that they are not related.
2. I think we can rename the first class here: 
<img width="748" alt="Снимок экрана 2023-06-01 в 13 57 58" src="https://github.com/python/cpython/assets/4660275/3b9f558f-5002-4e57-8673-3c1711d6a50b">

Right now the second class shadows the first one for no reason.
